src/main/java/com/example/rps/RPS.java

```diff
@@ -1,34 +1,33 @@
 package com.example.rps;
 
-import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+
 import java.util.HashMap;
 
 public class RPS {
 
-    public static String result;
-    RPSOpponentMove RPSOpponentMove = new RPSOpponentMove();
-    public static String opponentMove;
-    public static String[] rps = {"rock", "paper", "scissors"};
+    private static String result;
+    MoveGenerator moveGenerator = new MoveGenerator();
```

Comment:
Call the movegenerator in the rpsApp and validate the moves in there, so that RPS and movegenerator are completely decoupled.

Reply:
But tbh, nothing wrong this implementation either!